webapp/app.py

```diff
@@ -1333,67 +1338,60 @@ def serve_sitemap():
         This sitemap tracks changes in the template files and is generated
         dynamically on every new push to main.
         """
-        try:
-            sitemap_path = os.getcwd() + "/templates/sitemap_tree.xml"
-            directory_path = os.getcwd() + "/templates"
-            base_url = "https://canonical.com"
+        sitemap_path = os.getcwd() + "/templates/sitemap_tree.xml"
+        directory_path = os.getcwd() + "/templates"
+        base_url = "https://canonical.com"
+
+        # Validate the secret if its a POST request
+        if flask.request.method == "POST":
+            expected_secret = os.getenv("SITEMAP_SECRET")
+            provided_secret = flask.request.headers.get(
+                "Authorization", ""
+            ).replace("Bearer ", "")
+
+            if provided_secret != expected_secret:
+                logging.warning("Invalid secret provided")
+                return {"error": "Unauthorized"}, 401
+
+        # Generate sitemap if update request or if it doesn't exist
+        if flask.request.method == "POST" or not os.path.exists(sitemap_path):
```

Comment:
```suggestion
        if not os.path.exists(sitemap_path):
```
This condition should be met and caught above

Reply:
So is this a fallback in the case where the sitemap isn't generated on build? I wonder if it would be better to just fail the build in that case